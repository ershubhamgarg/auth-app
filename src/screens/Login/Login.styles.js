import { StyleSheet } from "react-native";
import { LIGHT_BLUE, SECONDARY_BLUE } from "../../constants/styleConstants";

export const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    padding: 20,
  },
  title: { color: LIGHT_BLUE, fontSize: 13, fontWeight: "500" },
  logo: { height: 120, width: 120 },
  button: { marginTop: 50, width: "100%" },
  emailContainer: { marginVertical: 30 },
  footer: { marginTop: 10 },
  signup: { color: SECONDARY_BLUE, textDecorationLine: "underline" },
  alignCenter: { alignItems: "center" },
});
