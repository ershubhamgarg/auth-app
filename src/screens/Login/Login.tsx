import { useNavigation } from "@react-navigation/native";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import { useEffect, useState } from "react";
import {
  Alert,
  Image,
  ImageBackground,
  KeyboardAvoidingView,
  Platform,
  Pressable,
  Text,
  View,
} from "react-native";
import Button from "../../components/Button/Button";
import Input from "../../components/Input/Input";
import { useAuth } from "../../context/AuthContext";
import { MainStackParamList } from "../../navigation/MainStack";
import { ERROR_MSG } from "./../../constants/errorConstants";
import { styles } from "./Login.styles";
export const Login = () => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [emailError, setEmailError] = useState("");
  const [passwordError, setPasswordError] = useState("");
  const { login, user, error, setErrors } = useAuth();
  const navigation =
    useNavigation<NativeStackNavigationProp<MainStackParamList>>();

  const handleLogin = () => {
    if (!email.length) {
      setEmailError(ERROR_MSG.EMAIL);
    }
    if (!password.length) {
      setPasswordError(ERROR_MSG.PASSWORD);
    }
    if (email && password) {
      const payload = {
        email,
        password,
      };
      login(payload);
    }
  };

  useEffect(() => {
    if (user) {
      navigation.navigate("Home");
    }
    if (error) {
      Alert.alert(error);
    }
    setErrors?.();
  }, [user, error]);

  const onEmailChange = (e: string) => {
    if (e.length) {
      setEmail(e);
      setEmailError("");
    } else {
      setEmailError(ERROR_MSG.EMAIL);
    }
  };

  const onPasswordChange = (e: string) => {
    if (e.length) {
      setPassword(e);
      setPasswordError("");
    } else {
      setPasswordError(ERROR_MSG.PASSWORD);
    }
  };

  const gotoSignup = () => {
    navigation.navigate("Signup");
  };

  return (
    <ImageBackground
      source={require("../../assets/images/back.png")}
      style={styles.container}
    >
      <KeyboardAvoidingView
        behavior={Platform.OS === "ios" ? "padding" : "height"}
      >
        <View style={styles.alignCenter}>
          <Image
            source={require("../../assets/logo/logo.png")}
            style={styles.logo}
          />
          <Text style={styles.title}>
            Please enter your Email and Password to log in
          </Text>
        </View>
        <Input
          testID="login-username"
          placeholder="Email"
          label="Email"
          keyboardType="email-address"
          value={email}
          containerStyle={styles.emailContainer}
          onChangeText={onEmailChange}
          error={emailError}
        />
        <Input
          testID="login-password"
          placeholder="Password"
          label="Password"
          isPassword
          value={password}
          onChangeText={onPasswordChange}
          error={passwordError}
        />
        <Button
          containerStyle={styles.button}
          label="Log in"
          onPress={handleLogin}
        />
        <View style={styles.alignCenter}>
          <Text style={styles.footer}>
            Don't have an account ?{" "}
            <Pressable onPress={gotoSignup}>
              <Text style={styles.signup}>Signup</Text>
            </Pressable>
          </Text>
        </View>
      </KeyboardAvoidingView>
    </ImageBackground>
  );
};
