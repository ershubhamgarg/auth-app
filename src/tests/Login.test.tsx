import { render } from "@testing-library/react-native";
import { Login } from "../screens/Login";

describe("Test login screeen", () => {
  test("matches snapshot", () => {
    const { toJSON } = render(<Login />);
    expect(toJSON()).toMatchSnapshot();
  });
});
