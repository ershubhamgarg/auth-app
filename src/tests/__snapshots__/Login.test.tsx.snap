// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Test login screeen matches snapshot 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    {
      "flex": 1,
      "justifyContent": "center",
      "padding": 20,
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../src/assets/images/back.png",
      }
    }
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    onLayout={[Function]}
    style={
      {
        "paddingBottom": 0,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../src/assets/logo/logo.png",
          }
        }
        style={
          {
            "height": 120,
            "width": 120,
          }
        }
      />
      <Text
        style={
          {
            "color": "#16476A",
            "fontSize": 13,
            "fontWeight": "500",
          }
        }
      >
        Please enter your Email and Password to log in
      </Text>
    </View>
    <View
      style={
        [
          {
            "width": "100%",
          },
          {
            "marginVertical": 30,
          },
        ]
      }
    >
      <View
        style={
          {
            "borderBottomWidth": 1,
            "borderColor": "#000000",
            "borderRadius": 4,
            "paddingTop": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#878787",
                "fontSize": 12,
                "marginBottom": 5,
              },
              undefined,
            ]
          }
        >
          Email
        </Text>
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <TextInput
            autoCapitalize="none"
            autoComplete="off"
            autoCorrect={false}
            keyboardType="email-address"
            onChangeText={[Function]}
            placeholder="Email"
            secureTextEntry={false}
            style={
              [
                {
                  "fontSize": 16,
                  "paddingVertical": 5,
                },
                {
                  "flex": 1,
                },
              ]
            }
            testID="login-username"
            value=""
          />
        </View>
      </View>
    </View>
    <View
      style={
        [
          {
            "width": "100%",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          {
            "borderBottomWidth": 1,
            "borderColor": "#000000",
            "borderRadius": 4,
            "paddingTop": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#878787",
                "fontSize": 12,
                "marginBottom": 5,
              },
              undefined,
            ]
          }
        >
          Password
        </Text>
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <TextInput
            autoCapitalize="none"
            autoComplete="off"
            autoCorrect={false}
            onChangeText={[Function]}
            placeholder="Password"
            secureTextEntry={true}
            style={
              [
                {
                  "fontSize": 16,
                  "paddingVertical": 5,
                },
                {
                  "flex": 1,
                },
              ]
            }
            testID="login-password"
            value=""
          />
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <Image
              resizeMode="contain"
              source={
                {
                  "testUri": "../../../src/assets/icons/hide.png",
                }
              }
              style={
                [
                  {
                    "height": 20,
                    "width": 20,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "#132440",
            "borderRadius": 5,
            "flexDirection": "row",
            "justifyContent": "center",
            "minHeight": 50,
            "padding": 10,
            "shadowOffset": {
              "height": 1,
              "width": 1,
            },
            "shadowOpacity": 0.4,
          },
          {
            "marginTop": 50,
            "width": "100%",
          },
        ]
      }
    >
      <Text
        style={
          {
            "color": "#ffffff",
            "fontSize": 15,
            "fontWeight": "700",
            "marginLeft": 10,
          }
        }
      >
        Log in
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <Text
        style={
          {
            "marginTop": 10,
          }
        }
      >
        Don't have an account ?
         
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <Text
            style={
              {
                "color": "#3B9797",
                "textDecorationLine": "underline",
              }
            }
          >
            Signup
          </Text>
        </View>
      </Text>
    </View>
  </View>
</View>
`;
